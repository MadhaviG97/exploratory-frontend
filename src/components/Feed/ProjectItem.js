import React from "react";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import Avatar from "@material-ui/core/Avatar";
import Typography from "@material-ui/core/Typography";
import { red } from "@material-ui/core/colors";

import { Box, Link, Divider, Button } from "@material-ui/core";
import BookmarkBorderIcon from "@material-ui/icons/BookmarkBorder";
import { getDateTime } from "../Chat/Utility/DTUtility";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      maxWidth: 600,
      margin: "10px",
    },
    media: {
      height: 0,
      paddingTop: "56.25%", // 16:9
    },
    avatar: {
      backgroundColor: red[500],
    },
  })
);

export default function ProjectItem({ ResearchItem }) {
  const classes = useStyles();
  console.log(ResearchItem);
  return (
    <Card className={classes.root}>
      <CardHeader
        avatar={
          <Avatar
            className={classes.avatar}
            src={ResearchItem.profile_picture}
          />
        }
        title={
          ResearchItem.first_name +
          " " +
          ResearchItem.last_name +
          " - " +
          ResearchItem.institution
        }
        subheader={
          ResearchItem.published_at
            ? "Published on: " + getDateTime(ResearchItem.published_at)
            : ""
        }
      />
      {ResearchItem.poster_image && ResearchItem.poster_image.length>0 ? (
        <CardMedia
          className={classes.media}
          image={ResearchItem.poster_image[0]}
        />
      ) : (
        <div></div>
      )}

      <CardContent>
        <Box fontWeight="fontWeightBold" color="secondary" component="h3">
          <Link href={"/project/viewproject/" + ResearchItem.id}>
            {ResearchItem.title}
          </Link>
        </Box>
        <Typography variant="body2" color="textSecondary" component="p">
          {ResearchItem.description.length > 500
            ? ResearchItem.description.slice(0, 500) + "..."
            : ResearchItem.description}
        </Typography>
      </CardContent>

      <Divider />

      <CardActions disableSpacing>
        <Button variant="outlined" startIcon={<BookmarkBorderIcon />}>
          Follow
        </Button>
      </CardActions>
    </Card>
  );
}
