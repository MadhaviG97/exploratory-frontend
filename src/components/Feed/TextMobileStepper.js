import React from 'react';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import MobileStepper from '@material-ui/core/MobileStepper';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';

const useStyles = makeStyles((theme) => ({
    root: {
        // maxWidth: 400,
        flexGrow: 1,
    },
    header: {
        display: 'flex',
        alignItems: 'center',
        height: 50,
        paddingLeft: theme.spacing(4),
        backgroundColor: theme.palette.background.default,
    },
    img: {
        // height: 255,
        // maxWidth: 400,
        maxHeight: 400,
        overflow: 'hidden',
        display: 'block',
        width: '100%',
    },
}));

export default function TextMobileStepper(props) {
    const classes = useStyles();
    const theme = useTheme();
    const [activeStep, setActiveStep] = React.useState(0);
    const maxSteps = props.images.length;

    const handleNext = () => {
        setActiveStep((prevActiveStep) => prevActiveStep + 1);
    };

    const handleBack = () => {
        setActiveStep((prevActiveStep) => prevActiveStep - 1);
    };

    return (
        <div className={classes.root}>

            <img
                className={classes.img}
                src={props.images[activeStep]}
                alt={"Project Image"}
            />
            <MobileStepper
                steps={maxSteps}
                position="static"
                variant="text"
                activeStep={activeStep}
                nextButton={
                    <Button size="small" onClick={handleNext} disabled={activeStep === maxSteps - 1}>
                        {theme.direction === 'rtl' ? <KeyboardArrowLeft /> : <KeyboardArrowRight />}
                    </Button>
                }
                backButton={
                    <Button size="small" onClick={handleBack} disabled={activeStep === 0}>
                        {theme.direction === 'rtl' ? <KeyboardArrowRight /> : <KeyboardArrowLeft />}
                    </Button>
                }
            />
        </div>
    );
}
