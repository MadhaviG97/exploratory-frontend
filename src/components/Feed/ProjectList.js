import React from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import ButtonBase from '@material-ui/core/ButtonBase';
import ProjectItem from './ProjectItem'
import Container from '@material-ui/core/Container';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
  }),
);


const ProjectList = (props) => {

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <React.Fragment>

      <Container maxWidth="sm" style={{backgroundColor: '#cfe8fc', padding:'10px'}}>

        { props.projects && props.projects.length >0 ? (
            props.projects.map(currentProject => (
              <ProjectItem ResearchItem={currentProject} />
            ))
          ):<div style={{paddingTop:"100px", margin:"auto"}}>No Projects</div>            
        }

      </Container>
      </React.Fragment>
  </div>
  );
}

export default ProjectList