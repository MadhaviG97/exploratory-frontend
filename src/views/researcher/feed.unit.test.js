import React from 'react';
import ReactDOM from 'react-dom';
import Feed from './feed';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
    user: { userData: {isAuth:true,first_name:'gamlath'} }
  });
jest.mock('react-router-dom', () => ({
    useHistory: () => ({
      push: jest.fn(),
    }),
    useLocation: () => ({
        push: jest.fn(),
      }),
  }));
describe('feed', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
        <Provider store={store}>
            <Feed />
        </Provider>, div);
    });
    it('renders correctly', () => {
        const tree = renderer
          .create(
          <Provider store={store}>
            <Feed />
          </Provider>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
});