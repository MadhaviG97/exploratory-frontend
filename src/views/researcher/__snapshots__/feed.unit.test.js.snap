// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feed renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-151"
>
  <div
    className="MuiBox-root MuiBox-root-152"
  >
    <div
      className="makeStyles-root-4"
    >
      <div
        className="MuiContainer-root MuiContainer-maxWidthSm"
        style={
          Object {
            "backgroundColor": "#cfe8fc",
            "padding": "10px",
          }
        }
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-13 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-avatar"
            >
              <div
                className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-15 MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiAvatar-fallback"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
              >
                Indika Perera - University of Moratuwa
              </span>
              <span
                className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Published on: 2020 02 20
              </span>
            </div>
          </div>
          <div
            className="MuiCardMedia-root makeStyles-media-14"
            style={
              Object {
                "backgroundImage": "url(\\"/images/tmp/food.jpg\\")",
              }
            }
          />
          <div
            className="MuiCardContent-root"
          >
            <h3
              className="MuiBox-root MuiBox-root-153"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/project/viewproject/1"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Introduction to Computer Architecture
              </a>
            </h3>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              When I say output, I mean not just output viewable to the end user as in the case of a dialog box, but also HTML output that will eventually be printed to a page with JavaScript. Note that HTML output is rendered by the browser, making any formatting of the HTML of no consequence to the end user. It is the scripter who benefits from cleanly formatted source code. When you create a plain HTML page and view it in your browser, if it doesn't look right, what do you do? You look at the source code. ...
            </p>
          </div>
          <hr
            className="MuiDivider-root"
          />
          <div
            className="MuiCardActions-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
                    />
                  </svg>
                </span>
                Follow
              </span>
            </button>
          </div>
        </div>
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-13 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-avatar"
            >
              <div
                className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-15 MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiAvatar-fallback"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
              >
                Indika Perera - University of Moratuwa
              </span>
              <span
                className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Published on: 2020 02 20
              </span>
            </div>
          </div>
          <div
            className="MuiCardMedia-root makeStyles-media-14"
            style={
              Object {
                "backgroundImage": "url(\\"/images/tmp/food.jpg\\")",
              }
            }
          />
          <div
            className="MuiCardContent-root"
          >
            <h3
              className="MuiBox-root MuiBox-root-154"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/project/viewproject/1"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Introduction to Computer Architecture
              </a>
            </h3>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              When I say output, I mean not just output viewable to the end user as in the case of a dialog box, but also HTML output that will eventually be printed to a page with JavaScript. Note that HTML output is rendered by the browser, making any formatting of the HTML of no consequence to the end user. It is the scripter who benefits from cleanly formatted source code. When you create a plain HTML page and view it in your browser, if it doesn't look right, what do you do? You look at the source code. ...
            </p>
          </div>
          <hr
            className="MuiDivider-root"
          />
          <div
            className="MuiCardActions-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
                    />
                  </svg>
                </span>
                Follow
              </span>
            </button>
          </div>
        </div>
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-13 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-avatar"
            >
              <div
                className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-15 MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiAvatar-fallback"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
              >
                Indika Perera - University of Moratuwa
              </span>
              <span
                className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Published on: 2020 02 20
              </span>
            </div>
          </div>
          <div
            className="MuiCardMedia-root makeStyles-media-14"
            style={
              Object {
                "backgroundImage": "url(\\"/images/tmp/food.jpg\\")",
              }
            }
          />
          <div
            className="MuiCardContent-root"
          >
            <h3
              className="MuiBox-root MuiBox-root-155"
            >
              <a
                className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/project/viewproject/1"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Introduction to Computer Architecture
              </a>
            </h3>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              When I say output, I mean not just output viewable to the end user as in the case of a dialog box, but also HTML output that will eventually be printed to a page with JavaScript. Note that HTML output is rendered by the browser, making any formatting of the HTML of no consequence to the end user. It is the scripter who benefits from cleanly formatted source code. When you create a plain HTML page and view it in your browser, if it doesn't look right, what do you do? You look at the source code. ...
            </p>
          </div>
          <hr
            className="MuiDivider-root"
          />
          <div
            className="MuiCardActions-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
                    />
                  </svg>
                </span>
                Follow
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
